node/modules/lp2p/pubsub.go

```diff
@@ -31,8 +43,132 @@ func PubsubTracer() PubsubOpt {
 }
 
 func GossipSub(pubsubOptions ...PubsubOpt) interface{} {
-	return func(mctx helpers.MetricsCtx, lc fx.Lifecycle, host host.Host) (service *pubsub.PubSub, err error) {
-		return pubsub.NewGossipSub(helpers.LifecycleCtx(mctx, lc), host, paresOpts(host, pubsubOptions)...)
+	return func(mctx helpers.MetricsCtx, lc fx.Lifecycle, host host.Host, nn dtypes.NetworkName, bp dtypes.BootstrapPeers) (service *pubsub.PubSub, err error) {
+		bootstrappers := make(map[peer.ID]struct{})
+		for _, pi := range bp {
+			bootstrappers[pi.ID] = struct{}{}
+		}
+		_, isBootstrapNode := bootstrappers[host.ID()]
+
+		v11Options := []pubsub.Option{
+			// Gossipsubv1.1 configuration
+			pubsub.WithFloodPublish(true),
+			pubsub.WithPeerScore(
+				&pubsub.PeerScoreParams{
+					AppSpecificScore: func(p peer.ID) float64 {
+						// return a heavy positive score for bootstrappers so that we don't unilaterally prune
+						// them and accept PX from them.
+						// we don't do that in the bootstrappers themselves to avoid creating a closed mesh
+						// between them (however we might want to consider doing just that)
+						_, ok := bootstrappers[p]
+						if ok && isBootstrapNode {
```

Comment:
should this be `if ok && !isBootstrapNode` instead?

Reply:
ah yes, indeed! Thanks for spotting that.

Reply:
fixed.